### PR TITLE
Fix study series QIDO bug

### PR DESCRIPTION
**Pull request by bcarthic**
## Description
Currently if there are 2 instances in a same study. QIDO with study series will fail.

https://localhost:63838/v2/studies/{StudyId}/series

QueryPlan

image

## Related issues
Addresses **[AB#114069].**

## Testing
New tests are added

### CODE REVIEW

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs [hunk 1]
```diff
@@ -210,6 +210,30 @@ await PostDicomFileAsync(new DicomDataset()
...
+    [Fact]
+    public async Task GivenSearchRequestWith2Instances_StudySeriesLevel_MatchResult()
+    {
+        // 2 instances in the same study, 2 series with 1 instance each CT and MRI
+        DicomDataset matchInstance = await PostDicomFileAsync(new DicomDataset()
+        {
+             { DicomTag.Modality, "CT" },
+        });
+        var studyId = matchInstance.GetSingleValue<string>(DicomTag.StudyInstanceUID);
+        var seriesId = matchInstance.GetSingleValue<string>(DicomTag.SeriesInstanceUID);
```
Comment (wsugarman) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:222]: I think you can remove `seriesId`

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs [hunk 2]
```diff
@@ -210,6 +210,30 @@ await PostDicomFileAsync(new DicomDataset()
...
+        {
+             { DicomTag.StudyInstanceUID, studyId },
+             { DicomTag.Modality, "MRI" }
+        });
+        using DicomWebAsyncEnumerableResponse<DicomDataset> response = await _client.QueryStudySeriesAsync(studyId, string.Empty);
+
+        DicomDataset[] datasets = await response.ToArrayAsync();
+
+        Assert.Equal(2, datasets.Length);
+        DicomDataset testDataResponse = datasets.FirstOrDefault(ds => ds.GetSingleValue<string>(DicomTag.StudyInstanceUID) == studyId);
```
Comment (wsugarman) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:233]:
```suggestion
        Assert.All(datasets, d => d.GetSingleValue<string>(DicomTag.StudyInstanceUID) == studyId);
```

**Review by wsugarman — APPROVED** [review]

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs (hunk 2 above)
Comment (wsugarman) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:233]: Then you can remove the last line

Comment (bcarthic) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:233]: Good idea, it needs to be like 

```
Assert.All(datasets, d => Assert.True(d.GetSingleValue<string>(DicomTag.StudyInstanceUID) == studyId));
```

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs (hunk 1 above)
Comment (bcarthic) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs]: Fixed

**Review by wsugarman — DISMISSED** [review]

**Review by wsugarman — APPROVED** [review]

**Pull request merged**